service/storage_service.cc

```diff
@@ -4656,8 +4656,10 @@ future<> storage_service::raft_check_and_repair_cdc_streams() {
         }
 
         curr_gen = _topology_state_machine._topology.current_cdc_generation_id;
-
-        // FIXME: check if the current generation is optimal, don't request new one if it isn't
+        auto gen = co_await _sys_ks.local().read_cdc_generation(curr_gen->id);
```

Comment:
Please add a sanity check for `curr_gen` being `nullopt`.
It should not happen but that's hard to prove so we should stay on the safe side.

If `curr_gen` is nullopt, print an `ERROR` log message and continue (i.e. we need to create a generation in this case).

Reply:
Done.